File: DynamicProgramming/746. Min Cost Climbing Stairs/Solution-DP-BottomUp.java
class Solution {
    public int minCostClimbingStairs(int[] cost) {
        int n = cost.length;

        int dpOne = cost[0];
        int dpTwo = cost[1];
        for (int i = 2; i < n; i++) {
            int current = cost[i] + Math.min(dpOne, dpTwo);
            dpOne = dpTwo;
            dpTwo = current;
        }
        return Math.min(dpOne, dpTwo);
    }
}

File: DynamicProgramming/746. Min Cost Climbing Stairs/Solution-DP-BottomUp.cpp
class Solution
{
public:
    int minCostClimbingStairs(vector<int> &cost)
    {
        int n = cost.size();
        int fir = cost[0];
        int sec = cost[1];

        for (int i = 2; i < n; i++)
        {
            int curr = cost[i] + min(fir, sec);
            fir = sec;
            sec = curr;
        }

        return min(fir, sec);
    }
};
